src/components/space/charts/ActionsByUserChart.js

```diff
@@ -0,0 +1,98 @@
+import React, { useContext } from 'react';
+import { useTranslation } from 'react-i18next';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import {
+  BarChart,
+  CartesianGrid,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Bar,
+  ResponsiveContainer,
+} from 'recharts';
+import EmptyChart from './EmptyChart';
+import {
+  getActionNumByVerb,
+  filterActionsByUser,
+  findYAxisMax,
+} from '../../../utils/api';
+import { COLORS, CONTAINER_HEIGHT } from '../../../config/constants';
+import { DataContext } from '../../context/DataProvider';
+import {
+  ACTIONS_BY_USER_MAX_DISPLAYED_USERS,
+  ACTION_TYPES,
+} from '../../../config/constants';
+
+const useStyles = makeStyles(() => ({
+  typography: { textAlign: 'center' },
+}));
+
+const ActionsByUserChart = () => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const { actions, selectedUsers, allMembers } = useContext(DataContext);
+  const users = selectedUsers.length ? selectedUsers : allMembers;
+  const yAxisMax = findYAxisMax(users);
+  let formattedActions = [];
+
+  users.forEach((user) => {
+    const actionsByVerb = getActionNumByVerb(
+      filterActionsByUser(actions, [{ id: user.id }]),
+    );
+    const userActions = {
+      id: user.id,
+      name: user.name,
+      total: actionsByVerb.total,
+    };
+    ACTION_TYPES.forEach((actionType) => {
+      if (actionsByVerb[actionType]) {
+        userActions[actionType] = actionsByVerb[actionType];
+      }
+    });
+    formattedActions.push(userActions);
+  });
+
+  const maxUsers = ACTIONS_BY_USER_MAX_DISPLAYED_USERS;
+  const title = 'Actions by Most Active Users';
+
+  // sort by total actions in descending order
+  formattedActions.sort((a, b) => b.total - a.total);
+  // get top 10 users
+  formattedActions = formattedActions.slice(0, maxUsers);
+  // filter out users with no actions
+  formattedActions = formattedActions.filter((user) => user.total);
+
+  if (formattedActions.length === 0) {
+    return <EmptyChart selectedUsers={selectedUsers} chartTitle={t(title)} />;
+  }
+
+  return (
+    <>
+      <Typography variant="h6" className={classes.typography}>
+        {t(title)}
+      </Typography>
+      <ResponsiveContainer width="95%" height={CONTAINER_HEIGHT}>
+        <BarChart
+          data={formattedActions}
+          margin={{ top: 30, bottom: 20, left: 20, right: 20 }}
+        >
+          <CartesianGrid strokeDasharray="2" />
+          <XAxis dataKey="name" tick={{ fontSize: 14 }} />
+          <YAxis tick={{ fontSize: 14 }} domain={[0, yAxisMax]} />
+          <Tooltip />
+          {ACTION_TYPES.map((actionType, index) => (
+            <Bar
+              key=""
```

Comment:
I think so, otherwise it prompts errors, or any other ways to fix it?
![image](https://user-images.githubusercontent.com/44131581/177789143-f143ae19-140c-4a83-a848-4adbf7ff99dc.png)

Reply:
The display would crash with actionType in key
<img width="575" alt="image" src="https://user-images.githubusercontent.com/44131581/178695398-f3d71619-48ce-4c69-88a3-f801eb15d6b6.png">

Reply:
huuum okay then we'll let it like this 🤷